day1/lib/day1.rb

```diff
@@ -0,0 +1,47 @@
+class Day1
+  attr_accessor :expense_report
+
+  def initialize(expense_report)
+    @expense_report = expense_report
+  end
+
+  def self.from_file(filepath)
+    expense_report_data = []
+    File.foreach(filepath).each do |line|
+      expense_report_data.append(line.chomp.to_i)
+    end
+    new(expense_report_data)
+  end
+
+  def calculate_answer(desired_sum)
+    element1, element2 = find_sum_elements(desired_sum)
+    if [element1, element2].all?
+      puts "Found #{element1} and #{element2} that sum to #{@desired_sum}"
+      element1 * element2
+    end
+  end
+
+  private
+
+  def find_sum_elements(desired_sum)
+    @expense_report.each_with_index do |line, index|
+      desired_element = desired_sum - line.to_i
+      all_others = @expense_report[(index + 1)..] + @expense_report[0...index]
```

Comment:
I looked and I looked and I just can't find a nice method for "the rest of the array you're iterating through", meaning all other elements other than the one we are currently in. Will have to settle for this.